### PR TITLE
shorten e2e_node test, fix pod ready test issue

### LINKED ISSUE #25703: Node E2E: Make conformance test run faster
Slow conformance test has caused issues like #25639 

@liangchenye We should make runtime/image conformance test run faster without breaking the test coverage, possibly by:
* Shorten consistently check interval #25695.
* Remove redundant test cases such as https://github.com/kubernetes/kubernetes/pull/24191#discussion_r62968819.
* ...

/cc @pwittrock

### PR DESCRIPTION
**Pull request by Random-Liu**
Fix #25703, #25812.
I fixed https://github.com/kubernetes/kubernetes/pull/26070 and open this PR.

@liangchenye 

[Analytics]()

### CODE REVIEW

File: test/e2e_node/runtime_conformance_test.go [hunk 1]
```diff
@@ -56,259 +56,129 @@ var _ = Describe("Container runtime Conformance Test", func() {
...
 	Describe("container runtime conformance blackbox test", func() {
-		var testCContainers []ConformanceContainer
 		namespace := "runtime-conformance"
 
-		BeforeEach(func() {
-			testCContainers = []ConformanceContainer{}
-		})
-
-		Context("when start a container that exits successfully", func() {
+		Context("when start a container that exits", func() {
```
Comment (pwittrock) [inline test/e2e_node/runtime_conformance_test.go]: nit: "when starting a container..."

File: test/e2e_node/container.go [hunk 2]
```diff
@@ -134,10 +149,10 @@ func (cc *ConformanceContainer) Present() (bool, error) {
 	return false, err
 }
 
-type ContainerState uint32
+type ContainerState int
```
Comment (dchen1107) [inline test/e2e_node/container.go]: why we need this change?

File: test/e2e_node/runtime_conformance_test.go [hunk 3]
```diff
@@ -56,259 +56,129 @@ var _ = Describe("[FLAKY] Container runtime Conformance Test", func() {
...
-				err := invalidImageContainer.Create()
-				Expect(err).NotTo(HaveOccurred())
-				testCContainers = append(testCContainers, invalidImageContainer)
+				Expect(invalidImageContainer.Create()).To(Succeed())
+				defer invalidImageContainer.Delete()
 
-				var status api.ContainerStatus
-				var phase api.PodPhase
+				Eventually(invalidImageContainer.GetPhase, retryTimeout, pollInterval).Should(Equal(testCase.Phase))
+				Consistently(invalidImageContainer.GetPhase, consistentCheckTimeout, pollInterval).Should(Equal(testCase.Phase))
```
Comment (pwittrock) [inline test/e2e_node/runtime_conformance_test.go]: Nice check that it stabilizes.

File: test/e2e_node/runtime_conformance_test.go [hunk 4]
```diff
@@ -56,259 +56,129 @@ var _ = Describe("[FLAKY] Container runtime Conformance Test", func() {
...
-				for _, testStatus := range testStatuses {
+				for _, testCase := range testCases {
 					tmpFile, err := ioutil.TempFile("", "restartCount")
 					Expect(err).NotTo(HaveOccurred())
 					defer os.Remove(tmpFile.Name())
 
-					// It fails in the first three runs and succeeds after that.
-					tmpCmd := fmt.Sprintf("echo 'hello' >> /restart-count/%s ; test $(wc -l /restart-count/%s| awk {'print $1'}) -ge %d", path.Base(tmpFile.Name()), path.Base(tmpFile.Name()), testCount+1)
-					testContainer.Name = testStatus.Name
+					// It fails in the first 2 runs and succeeds after that.
```
Comment (dchen1107) [inline test/e2e_node/runtime_conformance_test.go]: Based on the cmdline script here, looks it only failed first time, then second one it runs? Or misread this?

Comment (dchen1107) [inline test/e2e_node/runtime_conformance_test.go]: Should be: Failed in the first run, then succeeded at the second one, then run forever?

**Comment by dchen1107** [comment]
LGTM overall.

File: test/e2e_node/container.go (hunk 2 above)
Comment (Random-Liu) [inline test/e2e_node/container.go]: @dchen1107 In fact this is not necessary. Before we used bit operations, now we do not, then I changed `uint32` to `int`.

File: test/e2e_node/runtime_conformance_test.go (hunk 4 above)
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: ACK

File: test/e2e_node/runtime_conformance_test.go [hunk 5]
```diff
@@ -56,259 +56,129 @@ var _ = Describe("[FLAKY] Container runtime Conformance Test", func() {
...
+count=$(echo 'hello' >> $f ; wc -l $f | awk {'print $1'})
+if [ $count -eq 1 ]; then
+	exit 1
+elif [ $count -eq 2 ]; then
+	exit 0
+else
+	while true; do sleep 1; done
+	exit 0
+fi`
+					tmpCmd := fmt.Sprintf(cmdScripts, restartCountVolumePath+"/"+path.Base(tmpFile.Name()))
```
Comment (pwittrock) [inline test/e2e_node/runtime_conformance_test.go]: nit: You can also use paths.Join()

File: test/e2e_node/runtime_conformance_test.go (hunk 4 above)
Comment (pwittrock) [inline test/e2e_node/runtime_conformance_test.go]: s/fails/exits/
s/succeeds/continues to run/

File: test/e2e_node/runtime_conformance_test.go (hunk 5 above)
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: ACK

File: test/e2e_node/runtime_conformance_test.go [hunk 6]
```diff
@@ -56,259 +56,129 @@ var _ = Describe("[FLAKY] Container runtime Conformance Test", func() {
...
+					cmdScripts := `
+f=%s
+count=$(echo 'hello' >> $f ; wc -l $f | awk {'print $1'})
+if [ $count -eq 1 ]; then
+	exit 1
+elif [ $count -eq 2 ]; then
+	exit 0
+else
+	while true; do sleep 1; done
+	exit 0
```
Comment (pwittrock) [inline test/e2e_node/runtime_conformance_test.go]: nit: this exit is a no-op

File: test/e2e_node/runtime_conformance_test.go [hunk 7]
```diff
@@ -56,259 +56,129 @@ var _ = Describe("[FLAKY] Container runtime Conformance Test", func() {
...
-					testContainer.Name = testStatus.Name
+					// It fails in the first 2 runs and succeeds after that.
+					cmdScripts := `
+f=%s
+count=$(echo 'hello' >> $f ; wc -l $f | awk {'print $1'})
+if [ $count -eq 1 ]; then
+	exit 1
+elif [ $count -eq 2 ]; then
+	exit 0
+else
```
Comment (pwittrock) [inline test/e2e_node/runtime_conformance_test.go]: nit: don't need the else's

if ...
  exit 1
if ...
  exit 0
while true; do sleep 1; done

File: test/e2e_node/runtime_conformance_test.go (hunk 6 above)
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: ACK.

File: test/e2e_node/runtime_conformance_test.go (hunk 7 above)
Comment (Random-Liu) [inline test/e2e_node/runtime_conformance_test.go]: ACK.

**Comment by Random-Liu** [comment]
@dchen1107 @pwittrock Addressed comments and squashed.

**Comment by pwittrock** [comment]
Looks good.

**Pull request merged**